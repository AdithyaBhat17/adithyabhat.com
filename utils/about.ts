export const companies = [
  {
    name: 'Daikin',
    path: '/static/daikin.svg',
  },
  {
    name: 'Culture Plus',
    path: '/static/culture_plus.svg',
  },
  {
    name: 'Softway',
    path: '/static/softway.svg',
  },
  {
    name: 'Dish',
    path: '/static/dish.svg',
  },
  {
    name: 'Betsol',
    path: '/static/betsol_logo.svg',
  },
  {
    name: 'Homero',
    path: '/static/Homero.svg',
  },
]
