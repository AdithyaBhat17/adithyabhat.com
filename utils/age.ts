export function calculateAge(currentDate = new Date()): number {
  const birthDate = new Date('1998-01-17')

  let age = currentDate.getFullYear() - birthDate.getFullYear()

  if (
    currentDate.getMonth() < birthDate.getMonth() ||
    (currentDate.getMonth() === birthDate.getMonth() &&
      currentDate.getDate() < birthDate.getDate())
  ) {
    age = age - 1
  }

  return age
}
