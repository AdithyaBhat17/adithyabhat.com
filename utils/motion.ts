export const easing = [0.6, -0.05, 0.01, 0.99]

export const fadeInUp = {
  initial: {
    y: 20,
    opacity: 0,
    transition: { duration: 0.6, ease: easing },
  },
  animate: {
    y: 0,
    opacity: 1,
    transition: {
      duration: 0.6,
      ease: easing,
    },
  },
}

export const stagger = {
  animate: {
    transition: {
      staggerChildren: 0.1,
      delayChildren: 0.3,
    },
  },
}
