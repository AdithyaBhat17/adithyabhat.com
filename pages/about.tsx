import Head from '@/components/header'
import Container from '@/components/container'
import Link from 'next/link'
import { companies } from '@/utils/about'
import { motion } from 'framer-motion'
import { fadeInUp } from '@/utils/motion'
import useResetScroll from 'hooks/useResetScroll'
import Image from 'next/image'
import { calculateAge } from '@/utils/age'

export default function About({ companies, age }) {
  useResetScroll()
  return (
    <motion.div initial="initial" animate="animate" exit={{ opacity: 0 }}>
      <Head title="About - Adithya NR" />
      <Container className="2xl:px-96">
        <motion.div variants={fadeInUp} className="px-0 md:px-16 xl:px-32">
          <div className="flex justify-center">
            <Image
              width={250}
              height={250}
              loading="eager"
              className="block mx-auto"
              src="/static/about_hero.svg"
              alt="About Adithya"
            />
          </div>
          <h1 className="text-3xl md:text-4xl my-10 text-left sm:text-center font-semibold">
            About me
          </h1>
          <p className=" my-5 leading-8">
            As you might have guessed already, I&apos;m Adithya, a{' '}
            <span data-testid="age">{age}</span>
            -year-old UX Engineer from Bengaluru, India.
          </p>
          <p className="my-5 leading-8">
            I&apos;m a React nanodegree graduate and an IDF certified UX
            Designer. Free time is hard to come by, but I freelance and craft
            web apps for a living when it does.
          </p>
          <h2 className="text-md leading-8 uppercase opacity-50 mt-5 mb-3 font-semibold">
            Background
          </h2>
          <p className="my-5 leading-8">
            I studied computer science and engineering at RNS Institute of
            Technology, Bangalore. During my time at RNSIT, I volunteered as a
            graphic designer with a small team that helped me jump-start and
            hone my digital designer skills.
          </p>
          <p className="my-5 leading-8">
            I&apos;m currently working on the{' '}
            <a
              href="https://www.softway.com/b2e"
              target="_blank"
              rel="noopener noreferrer"
              className="left text-blue-800 font-medium"
            >
              Culture+â„¢
            </a>{' '}
            product suite at{' '}
            <a
              href="https://www.softway.com/"
              target="_blank"
              rel="noopener noreferrer"
              className="left text-blue-800 font-medium"
            >
              Softway
            </a>{' '}
            as a UX Engineer. Besides design, I also build web apps using React,
            Node, and many more tools.
          </p>
          <h3 className="text-xl md:text-3xl my-20 text-left sm:text-center font-semibold">
            Companies I&apos;ve worked with
          </h3>
          <div className="flex flex-wrap mx-auto justify-between sm:justify-around items-center mb-5">
            {companies?.map((company) => (
              <div key={company.name} className="w-1/2 md:w-1/5 mx-auto my-5">
                <img
                  loading="lazy"
                  className="mx-auto mb-10 w-1/2 sm:w-2/3 "
                  src={company.path}
                  alt={company.name}
                  title={company.name}
                />
              </div>
            ))}
          </div>
          <h3 className="text-xl md:text-3xl mt-20 mb-16 text-left sm:text-center font-semibold">
            Things I&apos;m good at
          </h3>
          <img
            loading="lazy"
            className="mx-auto w-3/4 md:w-1/3 block h-56 sm:h-auto"
            src="/static/skills.svg"
            alt="Things I'm good at"
          />
          <div className="flex flex-wrap mx-auto text-left md:text-center mt-16 md:mt-24">
            <div className="w-full mx-auto sm:flex-1">
              <h2 className="inline-flex items-center text-xl font-semibold">
                <img
                  src="/static/pen-tool.svg"
                  className="mr-5"
                  alt="Pen tool icon"
                />{' '}
                Design
              </h2>{' '}
              <br />
              <small className="block text-md sm:w-4/5 sm:mx-auto leading-7 my-5">
                User Experience, User Interface, Wireframing, Prototyping, Web
                design
              </small>
            </div>
            <div className="w-full mx-auto flex-1">
              <h2 className="inline-flex items-center text-xl font-semibold">
                <img
                  src="/static/terminal.svg"
                  className="mr-5"
                  alt="Terminal icon"
                />{' '}
                Development
              </h2>{' '}
              <br />
              <small className="block text-md sm:w-4/5 sm:mx-auto leading-7 my-5">
                JavaScript, Reactjs, Redux, Express, Nodejs, TypeScript, Nextjs,
                GraphQL, PostgreSQL
              </small>
            </div>
          </div>
          <h1 className="text-2xl md:text-5xl font-semibold my-20 md:my-40 leading-relaxed md:leading-relaxed">
            Got a project in mind, a question or something else? Get in touch
            with me{' '}
            <Link href="/contact">
              <a className="left"> here</a>
            </Link>
          </h1>
        </motion.div>
      </Container>
    </motion.div>
  )
}

export function getStaticProps() {
  const age = calculateAge()
  return {
    props: {
      companies,
      age,
    },
  }
}
