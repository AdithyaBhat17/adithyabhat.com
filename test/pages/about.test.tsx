import { render } from '../testUtils'
import About, { getStaticProps } from '@/pages/about'
import { companies } from '@/utils/about'
import { calculateAge } from '@/utils/age'

test('About page matches snapshot and displays my age correctly', () => {
  const age = calculateAge()
  const { container, getByTestId } = render(
    <About companies={companies} age={age} />
  )
  expect(container.innerHTML).toMatchSnapshot()
  expect(getByTestId('age').textContent).toBe(age.toString())
})

test('Get static props returns a list of companies', () => {
  const { props } = getStaticProps()
  expect(props.companies).toEqual(companies)
})
