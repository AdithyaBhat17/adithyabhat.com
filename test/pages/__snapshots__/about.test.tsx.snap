// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page matches snapshot and displays my age correctly 1`] = `"<div><div data-testid=\\"container\\" class=\\"px-6 md:px-36 lg:px-48 mb-6 2xl:px-96\\"><div class=\\"px-0 md:px-16 xl:px-32\\" style=\\"opacity: 0; transform: translateY(20px) translateZ(0);\\"><div class=\\"flex justify-center\\"><img alt=\\"About Adithya\\" loading=\\"eager\\" width=\\"250\\" height=\\"250\\" decoding=\\"async\\" data-nimg=\\"1\\" class=\\"block mx-auto\\" style=\\"color: transparent;\\" src=\\"/static/about_hero.svg\\"></div><h1 class=\\"text-3xl md:text-4xl my-10 text-left sm:text-center font-semibold\\">About me</h1><p class=\\" my-5 leading-8\\">As you might have guessed already, I'm Adithya, a UX Engineer from Bengaluru, India.</p><p class=\\"my-5 leading-8\\">For more than 5 years now, design has been the central piece of my world. On this quick and mind-blowing journey, I have moved over the years from being a graphic designer to a full-time UX Designer and UI Engineer.</p><p class=\\"my-5 leading-8\\">With a never-ending goal of maintaining a high standard for usability &amp; user experience, I have consistently worked with various startups and clients worldwide and helped them create unique and engaging digital products.</p><p class=\\"my-5 leading-8\\">I'm currently working on the Conversational CX platform at <a href=\\"http://yellow.ai\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"left\\">yellow.ai</a> as a Software Engineer (SDE-2).</p><h3 class=\\"text-xl md:text-3xl my-20 text-left sm:text-center font-semibold\\">Companies I've worked with</h3><div class=\\"grid grid-cols-2 md:grid-cols-3 lg:grid-cols-6 gap-10 lg:gap-14 justify-center items-center mb-5\\"><div class=\\"my-5\\"><img loading=\\"lazy\\" class=\\"\\" src=\\"/static/yellow_ai.svg\\" alt=\\"Yellow.ai\\" title=\\"Yellow.ai\\"></div><div class=\\"my-5\\"><img loading=\\"lazy\\" class=\\"\\" src=\\"/static/daikin.svg\\" alt=\\"Daikin\\" title=\\"Daikin\\"></div><div class=\\"my-5\\"><img loading=\\"lazy\\" class=\\"\\" src=\\"/static/softway.svg\\" alt=\\"Softway\\" title=\\"Softway\\"></div><div class=\\"my-5\\"><img loading=\\"lazy\\" class=\\"\\" src=\\"/static/dish.svg\\" alt=\\"Dish\\" title=\\"Dish\\"></div><div class=\\"my-5\\"><img loading=\\"lazy\\" class=\\"\\" src=\\"/static/betsol_logo.svg\\" alt=\\"Betsol\\" title=\\"Betsol\\"></div><div class=\\"my-5\\"><img loading=\\"lazy\\" class=\\"\\" src=\\"/static/Homero.svg\\" alt=\\"Homero\\" title=\\"Homero\\"></div></div><h3 class=\\"text-xl md:text-3xl mt-20 mb-16 text-left sm:text-center font-semibold\\">Things I'm good at</h3><img loading=\\"lazy\\" class=\\"mx-auto w-3/4 md:w-1/3 block h-56 sm:h-auto\\" src=\\"/static/skills.svg\\" alt=\\"Things I'm good at\\"><div class=\\"flex flex-wrap mx-auto text-left md:text-center mt-16 md:mt-24\\"><div class=\\"w-full mx-auto sm:flex-1\\"><h2 class=\\"inline-flex items-center text-xl font-semibold\\"><img src=\\"/static/pen-tool.svg\\" class=\\"mr-5\\" alt=\\"Pen tool icon\\"> Design</h2> <br><small class=\\"block text-md sm:w-4/5 sm:mx-auto leading-7 my-5\\">User Experience, User Interface, Wireframing, Prototyping, Web design</small></div><div class=\\"w-full mx-auto flex-1\\"><h2 class=\\"inline-flex items-center text-xl font-semibold\\"><img src=\\"/static/terminal.svg\\" class=\\"mr-5\\" alt=\\"Terminal icon\\"> Development</h2> <br><small class=\\"block text-md sm:w-4/5 sm:mx-auto leading-7 my-5\\">JavaScript, Reactjs, Redux, Express, Nodejs, TypeScript, Nextjs, GraphQL, PostgreSQL</small></div></div><h1 class=\\"text-2xl md:text-5xl font-semibold my-20 md:my-40 leading-relaxed md:leading-relaxed\\">Got a project in mind, a question or something else? Get in touch with me <a class=\\"left\\" href=\\"/contact\\"> here</a></h1></div></div></div>"`;
